### ISSUE TITLE
Package structure

### ISSUE THREAD

**Issue by DanielMDouglas** [tashutt/Gampy#19]
I'd like to provide this library as a package to analyzers (i.e. summer students at the moment) so that they can import the utilities into their own code without having to mess around in the library itself.  This essentially decouples the readout simulation and the analysis workflows.

This entails changing some of the directory structure and adding hooks for setuputils so that one can do something like `pip install [path to package (on filesystem or URL)]`.  At a later time, we can publish to pypi so that users can install from keyword or even a downstream package manager (`apt`, `pacman`, etc).  Users then can install the package once and in their analysis script they need only `from gampy.tools import electron_track_tools` or similar to access these functions.

I think this can be done without interrupting the interface too much, but will lower the barrier to entry for users who don't want to develop the code itself.

**Comment by DanielMDouglas** [comment]
See https://github.com/DUNE/larnd-sim for an example of this style of package structure

**Comment by bahrudint** [comment]
Good idea. The code I just merged to main has some of that structure. 
For example, to use gampy, you include:
```
from Gampy.tools import sims_tools, file_tools
from Gampy.performance_study_tools import monoenergetic_line_study as mls_study
```
the script has to be in the same directory as Gampy folder

**Comment by DanielMDouglas** [comment]
Please see what I'm working on in [package_structure_dev](https://github.com/tashutt/Gampy/tree/package_structure_dev)

**Comment by DanielMDouglas** [comment]
I've added a setup file which is used to install to your user/system path.  I've named the package `gampy` in keeping with [PEP-8](https://peps.python.org/pep-0008/#package-and-module-names).  I can import `gampy` from an arbitrary location, but I haven't tested anything rigorously yet (particularly if there are inter-library dependencies).  I've implemented the absolute import in `examples`, but it seems that many of the examples are currently broken...

**Comment by DanielMDouglas** [comment]
Can someone please point me to a known-good example that I can use to test this implementation?
